### PR TITLE
Add broker metric sample aggregator.

### PR DESCRIPTION
**Pull request by becketqin**

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/aggregator/KafkaBrokerMetricSampleAggregator.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig;
+import com.linkedin.kafka.cruisecontrol.monitor.metricdefinition.KafkaMetricDef;
+import com.linkedin.kafka.cruisecontrol.monitor.sampling.BrokerEntity;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The metrics sample aggregator for brokers.
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/aggregator/KafkaBrokerMetricSampleAggregator.java]: `private Set<PartitionEntity> allPartitions(Cluster cluster)` in `KafkaPartitionMetricSampleAggregator` may not return all partitions if there are offline partitions.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/MetricFetcherManager.java [hunk 2]
```diff
@@ -102,13 +108,15 @@ public MetricFetcherManager(KafkaCruiseControlConfig config,
...
                                KafkaPartitionMetricSampleAggregator partitionMetricSampleAggregator,
+                               KafkaBrokerMetricSampleAggregator brokerMetricSampleAggregator,
                                MetadataClient metadataClient,
                                MetricDef metricDef,
                                Time time,
                                MetricRegistry dropwizardMetricRegistry,
                                boolean createSampler) {
     _time = time;
     _partitionMetricSampleAggregator = partitionMetricSampleAggregator;
+    _brokerMetricSampleAggregator = brokerMetricSampleAggregator;
```
[the remark this answers is not in the document]
Comment (becketqin) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/MetricFetcherManager.java]: The existing sensors are actually shared by a single sampling fetcher. I changed the name to indicate that.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/aggregator/KafkaBrokerMetricSampleAggregator.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          config.getLong(KafkaCruiseControlConfig.BROKER_METRICS_WINDOW_MS_CONFIG),
+          config.getInt(KafkaCruiseControlConfig.MIN_SAMPLES_PER_BROKER_METRICS_WINDOW_CONFIG),
+          config.getInt(KafkaCruiseControlConfig.MAX_ALLOWED_EXTRAPOLATIONS_PER_BROKER_CONFIG),
+          config.getInt(KafkaCruiseControlConfig.BROKER_METRIC_SAMPLE_AGGREGATOR_COMPLETENESS_CACHE_SIZE_CONFIG),
+          KafkaMetricDef.brokerMetricDef());
+  }
+
+  public MetricSampleAggregationResult<String, BrokerEntity> aggregate(Set<BrokerEntity> brokerEntities) {
+    AggregationOptions<String, BrokerEntity>  aggregationOptions =
+        new AggregationOptions<>(0.0, 0.0, 1, brokerEntities,
```
[the remark this answers is not in the document]
Comment (becketqin) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/aggregator/KafkaBrokerMetricSampleAggregator.java]: The main reason is that the partitions are related, i.e. the metrics are supposed to be aggregated. Hence a necessary percentage of validity is required. For brokers, they are independent. An invalid broker won't invalidate another broker. This is why we don't need to expose the config.

**Comment by efeg** [comment]
LGTM.

**Pull request merged**